src/monitor/group_state_machine.c

```diff
@@ -406,6 +406,34 @@ ProceedGroupState(AutoFailoverNode *activeNode)
 		return true;
 	}
 
+	/*
+	 * when primary node is ready for replication:
+	 *  wait_standby -> catchingup
+	 *  primary -> apply_settings
+	 */
+	if (IsCurrentState(activeNode, REPLICATION_STATE_WAIT_STANDBY) &&
+		IsCurrentState(primaryNode, REPLICATION_STATE_PRIMARY))
+	{
+		char message[BUFSIZE];
+
+		LogAndNotifyMessage(
+			message, BUFSIZE,
+			"Setting goal state of " NODE_FORMAT
+			" to catchingup and " NODE_FORMAT
+			" to %s to edit synchronous_standby_names.",
+			NODE_FORMAT_ARGS(activeNode),
+			NODE_FORMAT_ARGS(primaryNode),
+			ReplicationStateGetName(primaryNode->reportedState));
+
+		/* start replication */
+		AssignGoalState(activeNode, REPLICATION_STATE_CATCHINGUP, message);
+
+		/* edit synchronous_standby_names to add the new standby now */
+		AssignGoalState(primaryNode, REPLICATION_STATE_APPLY_SETTINGS, message);
```

Comment:
Is this necessary when the node going to catchingup is a non replication quorum node?

Reply:
Playing the _Betteridge's law of headlines_ I see. Good move. I suppose we don't have to, and I'm testing that theory now, pushing soon.